pig/src/main/java/org/apache/zeppelin/pig/PigInterpreter.java

```diff
@@ -0,0 +1,150 @@
+
+  //Arguments to start pig with. More details available via 'pig -help'
+  static final String PIG_START_ARGS = "pig.start.args";
+  static final String DEFAULT_START_ARGS = "-useHCatalog -exectype tez";
+
+  //How long to wait before timing out (ms)
+  static final String PIG_TIMEOUT_MS = "pig.timeout.ms";
+  static final String DEFAULT_TIMEOUT_MS = "600000";
+
+  static {
+    Interpreter.register(
+      "pig",
+      "pig",
+      PigInterpreter.class.getName(),
+      new InterpreterPropertyBuilder()
+        .add(PIG_START_EXE, DEFAULT_START_EXE, "Pig executable used to start grunt shell")
+        .add(PIG_START_ARGS, DEFAULT_START_ARGS, "Starting arguments")
+        .add(PIG_TIMEOUT_MS, DEFAULT_TIMEOUT_MS, "Timeout (ms)")
+        .build()
+    );
+  }
+
+  public PigInterpreter(Properties property) {
+    super(property);
+  }
+
+  @Override
+  public void open() {}
+
+  @Override
+  public void close() {}
+
+
+  @Override
+  public InterpreterResult interpret(String cmd, InterpreterContext contextInterpreter) {
+    // use commandline to store string corresponding to pig shell command
+    // start with pig exectable name (or full path if provided)...
+    CommandLine cmdLine = CommandLine.parse(getProperty(PIG_START_EXE).trim());
+
+    // ...add any CLI arguments specified by user in interpreter settings
+    String startArgs = getProperty(PIG_START_ARGS).trim();
+    if (startArgs.length() > 0){
+      logger.info("Start arguments passed to pig: " + startArgs);
+      List<String> argList = Arrays.asList(startArgs.split("\\s+"));
+      for (String arg : argList) {
+        cmdLine.addArgument(arg, false);
+      }
+    }
+    // ...finally add contents of pig cell after the -e flag 
+    logger.info("Run pig command '" + cmd + "'");
+    long start = System.currentTimeMillis();
+    cmdLine.addArgument("-e", false);
+    cmdLine.addArgument(cmd, false);
+
+    // execute command and return success/failure based on its exit value
+    DefaultExecutor executor = new DefaultExecutor();
+    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+    executor.setStreamHandler(new PumpStreamHandler(outputStream));
+
+    int commandTimeOut = Integer.parseInt(getProperty(PIG_TIMEOUT_MS));
+    executor.setWatchdog(new ExecuteWatchdog(commandTimeOut));
+    try {
+      int exitValue = executor.execute(cmdLine);
+      return new InterpreterResult(InterpreterResult.Code.SUCCESS, outputStream.toString());
+    } catch (ExecuteException e) {
+      logger.error("Can not run " + cmd, e);
+      return new InterpreterResult(Code.ERROR, e.getMessage());
+    } catch (IOException e) {
+      logger.error("Can not run " + cmd, e);
+      return new InterpreterResult(Code.ERROR, e.getMessage());
+    }
+  }
+
+  @Override
+  public void cancel(InterpreterContext context) {}
+
+  @Override
+  public FormType getFormType() {
+    return FormType.SIMPLE;
+  }
+
+  @Override
+  public int getProgress(InterpreterContext context) {
```

Comment:
Thanks for all the feedback. Updated as below:
1. Documentation: added pig.md with interpreter documentation and added pig entry to index.md
2. Added test junit test based on passwd file parsing example here https://pig.apache.org/docs/r0.10.0/start.html#run
3. Removed author tag from comment (this was copied from shell interpreter https://github.com/apache/incubator-zeppelin/blob/master/shell/src/main/java/org/apache/zeppelin/shell/ShellInterpreter.java#L42)
4. Implemented cancel functionality
5. Display output stream in case of error